src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -195,7 +200,14 @@ public void apply(final VariantContext vc, final ReadsContext readsContext, fina
         final Set<String> rsIDs = getRSIDs(featureContext);
 
         if (VALIDATE_GVCF) {
-            genomeLocSortedSet.add(genomeLocSortedSet.getGenomeLocParser().createGenomeLoc(ref.getInterval().getContig(), ref.getInterval().getStart(), vc.getEnd()), true);
+            final SimpleInterval refInterval = ref.getInterval();
+            // Take advantage of adjacent blocks and just merge them so we dont have to keep so many objects in the set.
+            if (previousInterval != null && previousInterval.overlapsWithMargin(refInterval, 1)) {
+                genomeLocSortedSet.add(genomeLocSortedSet.getGenomeLocParser().createGenomeLoc(refInterval.getContig(), previousInterval.getEnd(), vc.getEnd()), true);
```

Comment:
I'm confused ad to why you are adding an interval in both cases of the if statment. I thought that the logic should be

```
if( contiguous)
  merge currentInterval into previous one
else
  add previous to set 
  replace previous with current
end
```

Specifically, only one branch add the interval. 

ofcourse, you still need to remember to add at the end..of the traversal. Are you doing that?